tldw_Server_API/app/api/v1/endpoints/admin/admin_system.py

```
@@ -27,6 +27,74 @@ async def get_security_alert_status() -> SecurityAlertStatusResponse:
+    user_id: int = Query(..., description="User ID to resolve permissions for"),
+    principal: AuthPrincipal = Depends(get_auth_principal),
+    db=Depends(get_db_transaction),
+) -> dict:
+    """Resolve effective permissions for a given user (debug tool)."""
+    return await admin_system_service.debug_resolve_permissions(user_id, db)
+
+
+@router.post("/debug/validate-token")
+async def debug_validate_token(
+    token: str = Query(..., description="JWT or API token to validate"),
+    principal: AuthPrincipal = Depends(get_auth_principal),
```

Ruff (0.15.7)

[warning] Unused function argument: `principal`

(ARG001)